src/controllers/RoomController.tsx

```diff
@@ -12,4 +12,60 @@ export const getRooms = async (property_id: Database['public']['Tables']['Proper
     }
 
     return res.data;
+}
+
+export const deleteRoom = async (room_id: number) => {
+    const { error } = await supabase
+        .from('Rooms')
+        .delete()
+        .eq('room_id', room_id);
+
+    if (error) {
+        throw (error);
+    }
+}
+
+export const createRooms = async (input: string, property_id: number) => {
+    //parse comma separated string
+    const names = input.split(',');
+    for (let i = 0; i < names.length; i++) {
+        names[i] = names[i].trim();
+    }
+
+    //get existing room names to avoid making duplicates later
+    let existingRooms = await getRooms(property_id);
+    let existingNames = new Set<string>();
+
+    //check all existing rooms
+    //if name is also in values retrieved from UI, add it to existing names
+    //if name not in values retrieved from UI, delete it from database
+    existingRooms.forEach(async room => {
+        if (names.includes(room.name)) {
+            existingNames.add(room.name);
+        }
+        else {
+            await deleteRoom(room.room_id);
+        }
+    })
+
+    //store new Room entry in database for each name and add room_id of each to returned array
+    let room_ids = await Promise.all(names.map(async name => {
+        if (name && !existingNames.has(name)) { //ensure room name isn't empty string or already used
+            const { data, error } = await supabase
+                .from('Rooms')
+                .insert({ name: name, property_id: property_id })
+                .select();
+
+            if (error) {
+                throw (error);
+            }
+
+            existingNames.add(data[0].name); //include newly created room names to avoid future duplicates
+
+            return data[0].room_id;
+        }
+    }
```

Comment:
Ah. This is why this method is returning an array containing either numbers or undefined. The mapping function returns nothing if the if statement on line 53 fails.

I'd recommend using a `.filter` method before the `.map` method to find only the names that should be created, then create them (and remove the if-statement within the mapping function). That way, the map will always return something.

Reply:
done